src/AssetConfig.php

```diff
@@ -525,6 +528,30 @@ public function addTarget($target, array $config)
     }
 
     /**
+     * Check the files list for any callbacks strings, executes them and replaces their
+     * position with the return array of the callback
+     *
+     * @param array $config Config data for the target.
+     * @return array $config Config data for the target.
+     */
+    protected function _applyCallbackProviders(array $config)
+    {
+        foreach ($config['files'] as $i => $file) {
+            if (preg_match(self::CALLBACK_PATTERN, $file, $matches)) {
+                $className = $matches[1];
+                $method = $matches[2];
+                $callbackFiles = call_user_func($className . '::' . $method);
+                if (is_array($callbackFiles)) {
+                    // Make sure we insert the files at the correct position, replacing
+                    // the callback string
+                    array_splice($config['files'], $i, 1, $callbackFiles);
+                }
+            }
+        }
+        return $config;
```

Comment:
Other expansions are currently happening in `MiniAsset\Factory::target()`. It current handles remote/local files, will soon handle glob expansion. I see this as similar to glob expansion so perhaps it belongs there?